tools/SetupFlow/DevHome.SetupFlow/Models/ConfigurationUnitResult.cs

```diff
@@ -38,6 +41,23 @@ public ConfigurationUnitResult(ElevatedConfigureUnitTaskResult result)
         ErrorDescription = result.ErrorDescription;
     }
 
+    public ConfigurationUnitResult(SDK.ApplyConfigurationUnitResult result)

```

Comment:
Is this a permanent SDK change?

Reply:
yea for the Dev Environments project we allow environments to be configured with WinGet configure by passing a config file as a string to the extension. However from an extensions point of view, there are times where they'd need to actually translate what they get from WinGet on a remote machine e.g a local VM or over the network back to the Dev Home extension. For example this can be done through strings, but now Dev Home and the extension need a common way to understand what is going on with the configuration (interpreting this string). So, we created a subset of WinGet interfaces/runtime classes in the SDK that extensions can then translate WinGet progress and results into, then send back to Dev Home via the SDK. From there once we receive it in Dev Home. We can reuse some of the classes we're using currently in the Configuration flow.

John from the WinGet team was on board with using a subset of the winget interfaces for this purpose. I can share the api review doc with you if you like.